### DISCUSSION TITLE
How to build on Raspberry PI

### DISCUSSION THREAD

**Discussion by timtuxworth** [mavlink-router/mavlink-router#384] [Q&A]
Sorry for being obtuse, but I don't understand how to build the mavlink-router. Yes I have read the readme, but that talks about building the dependancies and doesn't seem to explain how to build mavlink-router itself. Please help.

**Comment by lucasdemarchi** [comment]
meson setup build .
ninja -C build

**Reply by timtuxworth** [reply]
Thanks @lucasdemarchi  - I was confused because that part of the readme says "A typical configuration for a x86-64 system is shown below:" - not for a raspberry pi. I ran
```
sudo apt install git meson ninja-build
meson setup build .
ninja -C build
```
Ok - I did that and it seems to have completed ok, but where is the result?
Found it - in build/src (src?)

Wow - this thing runs great once you figure it out. Much better than mavproxy

**Reply by lucasdemarchi** [reply]
If you are building in the RPI itself , that is the correct instruction. We should probably reword the readme to say "native" instead of x86-64. People building for RPI often build in a desktop-like machine, for which case a cross-compiler is needed.

Result of the build is in the build directory.
